frontend/src/utilities/__tests__/clusterUtils.spec.ts

```diff
@@ -0,0 +1,118 @@
+import * as clusterInfo from '~/redux/selectors/clusterInfo';
+import { getOpenShiftConsoleServerURL, useOpenShiftURL } from '~/utilities/clusterUtils';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+let originalLocation: Location;
+
+jest.mock('~/utilities/const', () => ({
+  DEV_MODE: true, // Set to true
+}));
+
+jest.mock('~/redux/selectors/clusterInfo', () => ({
+  ...jest.requireActual('~/redux/selectors/clusterInfo'), // Use the actual module for other exports
+  useClusterInfo: jest.fn(),
+}));
+
+describe('getOpenShiftConsoleServerURL', () => {
+  beforeEach(() => {
+    originalLocation = { ...window.location };
+    Object.defineProperty(window, 'location', {
+      writable: true,
+      value: {
+        ...originalLocation,
+        hostname: 'localhost',
+        protocol: 'https:',
+        port: '443',
+      },
+    });
+  });
+
+  afterEach(() => {
+    // Restore window.location to its original state
+    Object.defineProperty(window, 'location', {
+      writable: true,
+      value: originalLocation,
+    });
+  });
+
+  it('should construct URL based on window location when no apiURL is provided', () => {
+    Object.defineProperty(window, 'location', {
+      writable: true,
+      value: {
+        ...originalLocation,
+        hostname: 'api.example.com',
+        protocol: 'https:',
+        port: '443',
+      },
+    });
+    const result = getOpenShiftConsoleServerURL();
+
+    // Assert the constructed URL
+    expect(result).toBe('https://console-openshift-console.example.com:443');
+  });
+  it('should construct URL with DEV_MODE true and provided apiURL', () => {
```

Comment:
@christianvogt How can we set DEV_MODE to `true` when we want to test this condition here https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/utilities/clusterUtils.ts#L9?

Reply:
```
jest.mock('~/utilities/const', () => ({
  get DEV_MODE() {
    return false;
  },
}));

const devModeMock = jest.spyOn(jest.requireMock('~/utilities/const'), 'DEV_MODE', 'get');

...
devModeMock.mockReturnValue(true);
// or
devModeMock.mockReturnValue(false);
```